proposals/0021-root-signature-in-clang.md

```diff
@@ -0,0 +1,315 @@
+        { i32 0, i32 0, i32 316 }, 
+        %"dxbc::RootSignature::ContainerRootParameter" 
+        { i32 1, i32 0, i32 348 }], 
+      [2 x %"dxbc::RootSignature::StaticSamplerDesc"]
+         [%"dxbc::RootSignature::StaticSamplerDesc" 
+           { i32 85, i32 1, i32 1, i32 1, float 0.000000e+00, i32 16, i32 4, 
+             i32 2, float 0.000000e+00, float 0x47EFFFFFE0000000, i32 1, 
+             i32 0, i32 0 }, 
+          %"dxbc::RootSignature::StaticSamplerDesc" 
+           { i32 21, i32 3, i32 1, i32 1, float 0.000000e+00, i32 16, i32 4, 
+           i32 2, float 0.000000e+00, float 0x47EFFFFFE0000000, i32 2, 
+           i32 0, i32 0 }], 
+      %1 { 
+        %"dxbc::RootSignature::ContainerRootDescriptor" 
+          { i32 0, i32 1, i32 8 }, 
+        %"dxbc::RootSignature::ContainerRootDescriptor" 
+          zeroinitializer, 
+        %"dxbc::RootSignature::ContainerRootDescriptor" 
+          zeroinitializer, 
+        %2 { %"dxbc::RootSignature::ContainerRootDescriptorTable" 
+        { i32 3, i32 244 }, 
+        [3 x %"dxbc::RootSignature::ContainerDescriptorRange"] 
+         [%"dxbc::RootSignature::ContainerDescriptorRange" 
+           { i32 2, i32 1, i32 1, i32 0, i32 0, i32 -1 }, 
+          %"dxbc::RootSignature::ContainerDescriptorRange" 
+           { i32 0, i32 8, i32 1, i32 0, i32 1, i32 -1 }, 
+          %"dxbc::RootSignature::ContainerDescriptorRange" 
+           { i32 1, i32 -1, i32 1, i32 0, i32 1, i32 -1 }] 
+       }, 
+       %3 {
+           %"dxbc::RootSignature::ContainerRootDescriptorTable" 
+           { i32 1, i32 324 }, 
+           [1 x %"dxbc::RootSignature::ContainerDescriptorRange"] 
+           [%"dxbc::RootSignature::ContainerDescriptorRange" 
+             { i32 3, i32 4, i32 0, i32 1, i32 0, i32 -1 }] }, 
+           %"dxbc::RootSignature::ContainerRootConstants" 
+             { i32 10, i32 0, i32 3 } 
+       }
+      }
+    }
+
+```
+
+* LLVM IR for root signature saved in metadata.
+
+```
+
+  ; named metadata for entry root signature
+  !hlsl.entry.rootsignatures = !{!2}
+  …
+  !2 = !{ptr @main, !3}
+
+  !3 = !{i32 2, i32 3, !4, !15}
+
+  !4 = !{!5, !6, !7, !8, !12, !14}
+
+  !5 = !{i32 2, i32 0, i32 0, i32 1, i32 8}
+
+  !6 = !{i32 3, i32 0, i32 0, i32 0, i32 0}
+
+  !7 = !{i32 4, i32 0, i32 0, i32 0, i32 0}
+
+  !8 = !{i32 0, i32 0, !9, !10, !11}
+  !9 = !{i32 2, i32 1, i32 1, i32 0, i32 0}
+  !10 = !{i32 0, i32 8, i32 1, i32 0, i32 1}
+  !11 = !{i32 1, i32 -1, i32 1, i32 0, i32 1}
+
+  !12 = !{i32 0, i32 0, !13}
+  !13 = !{i32 3, i32 4, i32 0, i32 1, i32 0}
+
+  !14 = !{i32 1, i32 0, i32 10, i32 0, i32 3}
+
+  !15 = !{!16, !17}
+  !16 = !{i32 85, i32 1, i32 1, i32 1, float 0.000000e+00, i32 16, i32 4, i32 2, float 0.000000e+00, float 0x47EFFFFFE0000000, i32 1, i32 0, i32 0}
+  !17 = !{i32 21, i32 3, i32 1, i32 1, float 0.000000e+00, i32 16, i32 4, i32 2, float 0.000000e+00, float 0x47EFFFFFE0000000, i32 2, i32 0, i32 0}
+
+```
+
+* Metadata format
```

Comment:
I think I understand what this is trying to show.  I suggest you add a column so we can see which line each of the rows in the table corresponds to.  eg:
Root Signature
| | Version | Flag | RootParameters | StaticSamplers |
|-| --- | --- | --- | --- |
| !3 | i32 2 | i32 3 | !4 | !15 |

Reply:
Fixed.